### PR TITLE
Update release checklist template

### PR DESCRIPTION
**Pull request by dgrisonnet**
* [ ] I added CHANGELOG entry for this change.
* [X] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/release.md [hunk 1]
```diff
@@ -16,6 +16,8 @@ about: Create release checklist
 - [ ] update downstream backport of [openshift/kube-rbac-proxy](https://github.com/openshift/kube-rbac-proxy)
 - [ ] update downstream backport of [openshift/k8s-prometheus-adapter](https://github.com/openshift/k8s-prometheus-adapter)
 - [ ] update downstream backport of [openshift/grafana](https://github.com/openshift/grafana)
+- [ ] update downstream backport of [openshift/prom-label-proxy](https://github.com/openshift/prom-label-proxy)
+- [ ] update downstream backport of [openshift/procfs](https://github.com/openshift/procfs)
```
Comment (s-urbaniak) [inline .github/ISSUE_TEMPLATE/release.md:20]: good points :+1:

**Comment by dgrisonnet** [comment]
@openshift/openshift-team-monitoring PTAL

File: .github/ISSUE_TEMPLATE/release.md (hunk 1 above)
Comment (simonpasquier) [inline .github/ISSUE_TEMPLATE/release.md:20]: This repository is a dependency of `github.com/openshift/node_exporter` so it should probably go with the `node_exporter` instructions. The workflow would be:
1. Bump `openshift/procfs` to the version used in the node_exporter's upstream version we want to bump to, including any downstream patch we may have.
2. Bump `openshift/node_exporter` to the upstream version replacing the `procfs` dependency by the downstream repo.

**Comment by simonpasquier** [comment]
we need also to bump versions in `hack/tools/go.mod` (especially prometheus version should be synced with the main `go.mod`). Noticed while looking at #1020 which fails because it uses the `group` aggregator that isn't supported by the `promtool` version pulled from `hack/tools/go.mod`.

**Review by simonpasquier — COMMENTED** [review]
/lgtm

**Pull request merged**